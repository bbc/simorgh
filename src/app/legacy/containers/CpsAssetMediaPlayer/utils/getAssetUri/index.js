const getAssetUri = assetUri => {
  if (assetUri.includes('newyddion')) {
    return assetUri.replace('newyddion', 'cymrufyw');
  }

  return assetUri;
};

export default getAssetUri;
