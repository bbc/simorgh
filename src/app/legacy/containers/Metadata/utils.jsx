import React from 'react';

export const getIconAssetUrl = (service, size) =>
  `https://static.files.bbci.co.uk/ws/simorgh-assets/public/${service}/images/icons/icon-${size}.png`;

const createIconLinks = (service, iconSizes, iconType) => {
  return iconSizes.map(size => {
    const iconAssetUrl = getIconAssetUrl(service, size);
    const key = `${service}-${size}`;

    if (iconType === 'icon') {
      return (
        <link
          key={key}
          rel="icon"
          type="image/png"
          href={iconAssetUrl}
          sizes={size}
        />
      );
    }

    return (
      <link key={key} rel="apple-touch-icon" sizes={size} href={iconAssetUrl} />
    );
  });
};

export const getIconLinks = (service, iconSizes) => {
  if (!iconSizes) {
    return null;
  }
  const iconTypes = Object.keys(iconSizes);
  return iconTypes.map(iconType => {
    return createIconLinks(service, iconSizes[iconType], iconType);
  });
};

export const getAppleTouchUrl = service => {
  return [
    process.env.SIMORGH_PUBLIC_STATIC_ASSETS_ORIGIN,
    process.env.SIMORGH_PUBLIC_STATIC_ASSETS_PATH,
    service,
    '/images/icons/icon-192x192.png',
  ].join('');
};

export const renderAlternateLinks = link => (
  <link
    rel="alternate"
    href={link.href}
    hrefLang={link.hrefLang}
    key={link.hrefLang}
  />
);

export const renderAppleItunesApp = ({
  iTunesAppId,
  canonicalLink,
  isAmp,
  hasAppleItunesAppBanner,
}) => {
  const isCanonical = !isAmp;

  const shouldRender = [
    iTunesAppId,
    canonicalLink,
    isCanonical,
    hasAppleItunesAppBanner,
  ].every(Boolean);

  if (shouldRender) {
    const content = `app-id=${iTunesAppId}, app-argument=${canonicalLink}?utm_medium=banner&utm_content=apple-itunes-app`;

    return <meta name="apple-itunes-app" content={content} />;
  }
  return null;
};
