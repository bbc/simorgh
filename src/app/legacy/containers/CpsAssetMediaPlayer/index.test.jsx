import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { RequestContextProvider } from '#contexts/RequestContext';
import { MEDIA_ASSET_PAGE } from '#app/routes/utils/pageTypes';
import {
  isNull,
  suppressPropWarnings,
} from '#psammead/psammead-test-helpers/src';
import { render } from '../../../components/react-testing-library-with-providers';
import { ServiceContextProvider } from '../../../contexts/ServiceContext';
import CpsAssetMediaPlayerContainer from '.';
import videoBlock from './fixtures';

const GenerateMediaPlayer = ({
  /* eslint-disable react/prop-types */
  platform,
  blocks,
  assetUri,
  /* eslint-enable react/prop-types */
}) => (
  <RequestContextProvider
    isAmp={platform === 'amp'}
    service="news"
    statusCode={200}
    platform={platform}
    id="foo"
    pageType={MEDIA_ASSET_PAGE}
    pathname="/pathname"
  >
    <ServiceContextProvider service="news">
      <BrowserRouter>
        <CpsAssetMediaPlayerContainer blocks={blocks} assetUri={assetUri} />
      </BrowserRouter>
    </ServiceContextProvider>
  </RequestContextProvider>
);

describe('MediaPlayer', () => {
  it('should render the canonical player without a placeholder', () => {
    const { container } = render(
      <GenerateMediaPlayer
        platform="canonical"
        blocks={[videoBlock]}
        assetUri="/pidgin/123456789"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render the amp player', () => {
    const { container } = render(
      <GenerateMediaPlayer
        platform="amp"
        blocks={[videoBlock]}
        assetUri="/pidgin/123456789"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render the canonical legacy media player', () => {
    const { container } = render(
      <GenerateMediaPlayer
        platform="canonical"
        blocks={[videoBlock]}
        assetUri="/russian/multimedia/2016/05/160505_v_diving_record"
        isLegacyMedia
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('should render the amp legacy media player', () => {
    const { container } = render(
      <GenerateMediaPlayer
        platform="amp"
        blocks={[videoBlock]}
        assetUri="/russian/multimedia/2016/05/160505_v_diving_record"
        isLegacyMedia
      />,
    );
    expect(container).toMatchSnapshot();
  });

  suppressPropWarnings(['assetUri']);
  isNull(
    'is Null when assetUri is not provided',
    <GenerateMediaPlayer platform="canonical" blocks={[videoBlock]} />,
  );
});
