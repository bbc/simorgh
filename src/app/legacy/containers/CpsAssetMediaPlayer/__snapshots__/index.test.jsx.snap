// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaPlayer should render the amp legacy media player 1`] = `
.emotion-1 {
  margin-top: 0.5rem;
}

@media (max-width: 14.9375rem) {
  .emotion-1 {
    margin-left: 0%;
  }
}

@media (min-width: 15rem) and (max-width: 24.9375rem) {
  .emotion-1 {
    margin-left: 0%;
  }
}

@media (min-width: 25rem) and (max-width: 37.4375rem) {
  .emotion-1 {
    margin-left: 0%;
  }
}

@media (min-width: 37.5rem) and (max-width: 62.9375rem) {
  .emotion-1 {
    margin-left: 0%;
  }
}

@media (min-width: 63rem) and (max-width: 79.9375rem) {
  .emotion-1 {
    margin-left: 16.666666666666668%;
  }
}

@media (min-width: 80rem) {
  .emotion-1 {
    margin-left: 33.333333333333336%;
  }
}

@supports (display: grid) {
  .emotion-1 {
    display: block;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 2;
    }
  }

  @media (min-width: 80rem) {
    .emotion-1 {
      grid-template-columns: repeat(12, 1fr);
      grid-column-end: span 12;
      grid-column-start: 5;
    }
  }
}

@media (max-width: 24.9375rem) {
  .emotion-1 {
    margin-top: 0.5rem;
  }
}

@media (min-width: 25rem) {
  .emotion-1 {
    margin-top: 1rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-1 {
    padding: 0 1rem;
  }
}

@media (min-width: 63rem) {
  .emotion-1 {
    padding: 0.5rem 0 0;
    margin-top: 2rem;
  }
}

.emotion-3 {
  margin: 0;
  padding-bottom: 1.5rem;
  width: 100%;
}

.emotion-5 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.emotion-7 {
  position: relative;
  height: 0;
  overflow: hidden;
  background-color: #F2F2F2;
  -webkit-background-position: center center;
  background-position: center center;
  background-repeat: no-repeat;
  -webkit-background-size: 60px 17px;
  background-size: 60px 17px;
  width: 100%;
  background-image: url(data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48c3ZnIHdpZHRoPSI0NzkiIGhlaWdodD0iMTM2IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPjxnIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCI+PGcgZmlsbD0iI0Q1RDBDRCI+PGc+PHBhdGggZD0iTTQ0OS41NTYgOTEuNTUzbC0yLjMxIDEuNDFjLTExLjE2NyA2LjgxOC0yMy4zMTMgMTAuNDc0LTM0LjM0NiAxMC40Ny0yMy42MS0uMDktMzkuMTYzLTE0LjA4My0zOS4yMjctMzQuNDUzLjAzLTE5LjkxOCAxNi4yNy0zNC42NjMgMzguNjMzLTM0LjcxOCAxMS4zODcuMDEgMjEuNzAzIDIuOTU0IDMzLjk2MiA5LjY3MmwyLjI1MSAxLjI0di0xOC4xOWwtLjk2Mi0uMzc3Yy0xMy44MjQtNS40NTgtMjQuNTMtNy44OS0zNS4xMDMtNy44ODQtMTYuMzQ2LS4wMDYtMzAuNTMzIDUuMzk0LTQwLjYzNyAxNC41NTctMTAuMTA1IDkuMTYzLTE2LjEwNiAyMi4xMDItMTYuMDk5IDM2Ljk1My4wMDggMTAuMzQ4IDQuMjc5IDIyLjQ4IDEzLjQyIDMyLjEwNSA5LjEyMSA5LjYyOCAyMy4xNjUgMTYuNjQ4IDQyLjQzIDE2LjYzOWguMDYzYzE1Ljk4IDAgMjcuMDYyLTMuNTYzIDM3LjA3NC04LjQ5MmwuODUxLS40MTRWOTEuNTUzek0zMzQgMTM1LjY5N2gxNDQuMTk1VjBIMzM0djEzNS42OTd6Ii8+PHBhdGggZD0iTTI3Ni45MzcgODkuOTY4Yy4wNDEtMTIuMzMtOC4xNzEtMjEuNjk2LTIxLjMwOC0yNS4zIDMuNTQ0LTEuODA5IDYuMzUtNC4wMjMgOC40MDQtNi43MjcgMi43NS0zLjYyMiA0LjA2MS04LjA2NCA0LjA0Ni0xMy4yMzUuMDE1LTYuMzU5LTIuNDg2LTEyLjgzOS03Ljg1OC0xNy42ODctNS4zNzItNC44NDctMTMuNTI2LTcuOTk3LTI0LjY1NC03Ljk5MUgyMDQuODN2OTcuNzI4aDM2LjA3M2MxMi44NyAwIDIxLjkwNi0zLjQ4MiAyNy43MjItOC42NSA1LjgxOC01LjE1NSA4LjMyLTExLjkxIDguMzEyLTE4LjEzOHpNMTY3IDEzNS42OThoMTQ0LjE5N1YwSDE2N3YxMzUuNjk3eiIvPjxwYXRoIGQ9Ik0xMDkuOTM3IDg5Ljk2OGMuMDQxLTEyLjMzLTguMTcxLTIxLjY5Ni0yMS4zMDgtMjUuMyAzLjU0NC0xLjgwOSA2LjM1LTQuMDIzIDguNDA0LTYuNzI3IDIuNzUtMy42MjIgNC4wNjEtOC4wNjQgNC4wNDYtMTMuMjM1LjAxNS02LjM1OS0yLjQ4Ni0xMi44MzktNy44NTgtMTcuNjg3LTUuMzcyLTQuODQ3LTEzLjUyNi03Ljk5Ny0yNC42NTQtNy45OTFIMzcuODN2OTcuNzI4aDM2LjA3M2MxMi44NyAwIDIxLjkwNi0zLjQ4MiAyNy43MjItOC42NSA1LjgxOC01LjE1NSA4LjMyLTExLjkxIDguMzEyLTE4LjEzOHpNMCAxMzUuNjk4aDE0NC4xOTdWMEgwdjEzNS42OTd6Ii8+PHBhdGggZD0iTTI1OC42NjIgODguMTk4Yy0uMDEzIDMuMjI5LTEuMDA3IDYuNDc1LTMuODk2IDkuMDExLTIuODg0IDIuNTM3LTcuODczIDQuNDYzLTE2LjEzMyA0LjQ2M0gyMjJWNzVoMTUuODkzYzcuNDExIDAgMTIuNjcgMS41MDIgMTUuOTY1IDMuODUgMy4yODkgMi4zNjIgNC43NzYgNS40NjMgNC44MDQgOS4zNDgiLz48cGF0aCBkPSJNOTEuNjYyIDg4LjE5OGMtLjAxMyAzLjIyOS0xLjAwNyA2LjQ3NS0zLjg5NiA5LjAxMS0yLjg4NCAyLjUzNy03Ljg3NCA0LjQ2My0xNi4xMzMgNC40NjNINTVWNzVoMTUuODkyYzcuNDEyIDAgMTIuNjcyIDEuNTAyIDE1Ljk2NiAzLjg1IDMuMjg5IDIuMzYyIDQuNzc2IDUuNDYzIDQuODA0IDkuMzQ4Ii8+PHBhdGggZD0iTTI0NS4xODYgNTUuNzljMy4wOTYtMi4yMzcgNC41OS01LjM4NiA0LjYxMy0xMC4xMjQtLjAxNS0zLjI1LS45NDMtNi4wMzMtMy4yODEtOC4xMTEtMi4zNDYtMi4wNzgtNi4zMy0zLjU1NS0xMi43NTQtMy41NTVIMjIydjI1LjI3NWg4LjA3NmM2Ljk4OC4wMDQgMTEuOTk4LTEuMjQzIDE1LjExLTMuNDg2Ii8+PHBhdGggZD0iTTc4LjE4NiA1NS43OWMzLjA5Ni0yLjIzNyA0LjU5LTUuMzg2IDQuNjEzLTEwLjEyNC0uMDE1LTMuMjUtLjk0My02LjAzMy0zLjI4Mi04LjExMUM3Ny4xNzIgMzUuNDc3IDczLjE4OCAzNCA2Ni43NjQgMzRINTV2MjUuMjc1aDguMDc2YzYuOTg4LjAwNCAxMS45OTgtMS4yNDMgMTUuMTEtMy40ODYiLz48L2c+PC9nPjwvZz48L3N2Zz4K);
}

@media (min-width: 25rem) {
  .emotion-7 {
    -webkit-background-size: 77px 22px;
    background-size: 77px 22px;
  }
}

@media (min-width: 63rem) {
  .emotion-7 {
    -webkit-background-size: 93px 27px;
    background-size: 93px 27px;
  }
}

<div>
  <div
    class="emotion-0 emotion-1 emotion-2"
    dir="ltr"
  >
    <div
      class="emotion-3 emotion-4"
    >
      <div
        class="emotion-5 emotion-6"
      >
        <amp-iframe
          allowfullscreen="allowfullscreen"
          data-e2e="media-player"
          frameborder="0"
          layout="fill"
          sandbox="allow-scripts allow-same-origin"
          scrolling="no"
          src="https://polling.test.bbc.co.uk/ws/av-embeds/cps/russian/multimedia/2016/05/160505_v_diving_record/p01kx42v/en-GB/amp"
          title="Media player"
        >
          <div
            class="emotion-7 emotion-8"
            data-e2e="image-placeholder"
            placeholder=""
            style="padding-bottom: 56.25%;"
          />
          <noscript />
        </amp-iframe>
      </div>
    </div>
  </div>
</div>
`;

exports[`MediaPlayer should render the amp player 1`] = `
.emotion-1 {
  margin-top: 0.5rem;
}

@media (max-width: 14.9375rem) {
  .emotion-1 {
    margin-left: 0%;
  }
}

@media (min-width: 15rem) and (max-width: 24.9375rem) {
  .emotion-1 {
    margin-left: 0%;
  }
}

@media (min-width: 25rem) and (max-width: 37.4375rem) {
  .emotion-1 {
    margin-left: 0%;
  }
}

@media (min-width: 37.5rem) and (max-width: 62.9375rem) {
  .emotion-1 {
    margin-left: 0%;
  }
}

@media (min-width: 63rem) and (max-width: 79.9375rem) {
  .emotion-1 {
    margin-left: 16.666666666666668%;
  }
}

@media (min-width: 80rem) {
  .emotion-1 {
    margin-left: 33.333333333333336%;
  }
}

@supports (display: grid) {
  .emotion-1 {
    display: block;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 2;
    }
  }

  @media (min-width: 80rem) {
    .emotion-1 {
      grid-template-columns: repeat(12, 1fr);
      grid-column-end: span 12;
      grid-column-start: 5;
    }
  }
}

@media (max-width: 24.9375rem) {
  .emotion-1 {
    margin-top: 0.5rem;
  }
}

@media (min-width: 25rem) {
  .emotion-1 {
    margin-top: 1rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-1 {
    padding: 0 1rem;
  }
}

@media (min-width: 63rem) {
  .emotion-1 {
    padding: 0.5rem 0 0;
    margin-top: 2rem;
  }
}

.emotion-3 {
  margin: 0;
  padding-bottom: 1.5rem;
  width: 100%;
}

.emotion-5 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.emotion-7 {
  position: relative;
  height: 0;
  overflow: hidden;
  background-color: #F2F2F2;
  -webkit-background-position: center center;
  background-position: center center;
  background-repeat: no-repeat;
  -webkit-background-size: 60px 17px;
  background-size: 60px 17px;
  width: 100%;
  background-image: url(data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48c3ZnIHdpZHRoPSI0NzkiIGhlaWdodD0iMTM2IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPjxnIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCI+PGcgZmlsbD0iI0Q1RDBDRCI+PGc+PHBhdGggZD0iTTQ0OS41NTYgOTEuNTUzbC0yLjMxIDEuNDFjLTExLjE2NyA2LjgxOC0yMy4zMTMgMTAuNDc0LTM0LjM0NiAxMC40Ny0yMy42MS0uMDktMzkuMTYzLTE0LjA4My0zOS4yMjctMzQuNDUzLjAzLTE5LjkxOCAxNi4yNy0zNC42NjMgMzguNjMzLTM0LjcxOCAxMS4zODcuMDEgMjEuNzAzIDIuOTU0IDMzLjk2MiA5LjY3MmwyLjI1MSAxLjI0di0xOC4xOWwtLjk2Mi0uMzc3Yy0xMy44MjQtNS40NTgtMjQuNTMtNy44OS0zNS4xMDMtNy44ODQtMTYuMzQ2LS4wMDYtMzAuNTMzIDUuMzk0LTQwLjYzNyAxNC41NTctMTAuMTA1IDkuMTYzLTE2LjEwNiAyMi4xMDItMTYuMDk5IDM2Ljk1My4wMDggMTAuMzQ4IDQuMjc5IDIyLjQ4IDEzLjQyIDMyLjEwNSA5LjEyMSA5LjYyOCAyMy4xNjUgMTYuNjQ4IDQyLjQzIDE2LjYzOWguMDYzYzE1Ljk4IDAgMjcuMDYyLTMuNTYzIDM3LjA3NC04LjQ5MmwuODUxLS40MTRWOTEuNTUzek0zMzQgMTM1LjY5N2gxNDQuMTk1VjBIMzM0djEzNS42OTd6Ii8+PHBhdGggZD0iTTI3Ni45MzcgODkuOTY4Yy4wNDEtMTIuMzMtOC4xNzEtMjEuNjk2LTIxLjMwOC0yNS4zIDMuNTQ0LTEuODA5IDYuMzUtNC4wMjMgOC40MDQtNi43MjcgMi43NS0zLjYyMiA0LjA2MS04LjA2NCA0LjA0Ni0xMy4yMzUuMDE1LTYuMzU5LTIuNDg2LTEyLjgzOS03Ljg1OC0xNy42ODctNS4zNzItNC44NDctMTMuNTI2LTcuOTk3LTI0LjY1NC03Ljk5MUgyMDQuODN2OTcuNzI4aDM2LjA3M2MxMi44NyAwIDIxLjkwNi0zLjQ4MiAyNy43MjItOC42NSA1LjgxOC01LjE1NSA4LjMyLTExLjkxIDguMzEyLTE4LjEzOHpNMTY3IDEzNS42OThoMTQ0LjE5N1YwSDE2N3YxMzUuNjk3eiIvPjxwYXRoIGQ9Ik0xMDkuOTM3IDg5Ljk2OGMuMDQxLTEyLjMzLTguMTcxLTIxLjY5Ni0yMS4zMDgtMjUuMyAzLjU0NC0xLjgwOSA2LjM1LTQuMDIzIDguNDA0LTYuNzI3IDIuNzUtMy42MjIgNC4wNjEtOC4wNjQgNC4wNDYtMTMuMjM1LjAxNS02LjM1OS0yLjQ4Ni0xMi44MzktNy44NTgtMTcuNjg3LTUuMzcyLTQuODQ3LTEzLjUyNi03Ljk5Ny0yNC42NTQtNy45OTFIMzcuODN2OTcuNzI4aDM2LjA3M2MxMi44NyAwIDIxLjkwNi0zLjQ4MiAyNy43MjItOC42NSA1LjgxOC01LjE1NSA4LjMyLTExLjkxIDguMzEyLTE4LjEzOHpNMCAxMzUuNjk4aDE0NC4xOTdWMEgwdjEzNS42OTd6Ii8+PHBhdGggZD0iTTI1OC42NjIgODguMTk4Yy0uMDEzIDMuMjI5LTEuMDA3IDYuNDc1LTMuODk2IDkuMDExLTIuODg0IDIuNTM3LTcuODczIDQuNDYzLTE2LjEzMyA0LjQ2M0gyMjJWNzVoMTUuODkzYzcuNDExIDAgMTIuNjcgMS41MDIgMTUuOTY1IDMuODUgMy4yODkgMi4zNjIgNC43NzYgNS40NjMgNC44MDQgOS4zNDgiLz48cGF0aCBkPSJNOTEuNjYyIDg4LjE5OGMtLjAxMyAzLjIyOS0xLjAwNyA2LjQ3NS0zLjg5NiA5LjAxMS0yLjg4NCAyLjUzNy03Ljg3NCA0LjQ2My0xNi4xMzMgNC40NjNINTVWNzVoMTUuODkyYzcuNDEyIDAgMTIuNjcyIDEuNTAyIDE1Ljk2NiAzLjg1IDMuMjg5IDIuMzYyIDQuNzc2IDUuNDYzIDQuODA0IDkuMzQ4Ii8+PHBhdGggZD0iTTI0NS4xODYgNTUuNzljMy4wOTYtMi4yMzcgNC41OS01LjM4NiA0LjYxMy0xMC4xMjQtLjAxNS0zLjI1LS45NDMtNi4wMzMtMy4yODEtOC4xMTEtMi4zNDYtMi4wNzgtNi4zMy0zLjU1NS0xMi43NTQtMy41NTVIMjIydjI1LjI3NWg4LjA3NmM2Ljk4OC4wMDQgMTEuOTk4LTEuMjQzIDE1LjExLTMuNDg2Ii8+PHBhdGggZD0iTTc4LjE4NiA1NS43OWMzLjA5Ni0yLjIzNyA0LjU5LTUuMzg2IDQuNjEzLTEwLjEyNC0uMDE1LTMuMjUtLjk0My02LjAzMy0zLjI4Mi04LjExMUM3Ny4xNzIgMzUuNDc3IDczLjE4OCAzNCA2Ni43NjQgMzRINTV2MjUuMjc1aDguMDc2YzYuOTg4LjAwNCAxMS45OTgtMS4yNDMgMTUuMTEtMy40ODYiLz48L2c+PC9nPjwvZz48L3N2Zz4K);
}

@media (min-width: 25rem) {
  .emotion-7 {
    -webkit-background-size: 77px 22px;
    background-size: 77px 22px;
  }
}

@media (min-width: 63rem) {
  .emotion-7 {
    -webkit-background-size: 93px 27px;
    background-size: 93px 27px;
  }
}

<div>
  <div
    class="emotion-0 emotion-1 emotion-2"
    dir="ltr"
  >
    <div
      class="emotion-3 emotion-4"
    >
      <div
        class="emotion-5 emotion-6"
      >
        <amp-iframe
          allowfullscreen="allowfullscreen"
          data-e2e="media-player"
          frameborder="0"
          layout="fill"
          sandbox="allow-scripts allow-same-origin"
          scrolling="no"
          src="https://polling.test.bbc.co.uk/ws/av-embeds/cps/pidgin/123456789/p01kx42v/en-GB/amp"
          title="Media player"
        >
          <div
            class="emotion-7 emotion-8"
            data-e2e="image-placeholder"
            placeholder=""
            style="padding-bottom: 56.25%;"
          />
          <noscript />
        </amp-iframe>
      </div>
    </div>
  </div>
</div>
`;

exports[`MediaPlayer should render the canonical legacy media player 1`] = `
.emotion-1 {
  margin-top: 0.5rem;
}

@media (max-width: 14.9375rem) {
  .emotion-1 {
    margin-left: 0%;
  }
}

@media (min-width: 15rem) and (max-width: 24.9375rem) {
  .emotion-1 {
    margin-left: 0%;
  }
}

@media (min-width: 25rem) and (max-width: 37.4375rem) {
  .emotion-1 {
    margin-left: 0%;
  }
}

@media (min-width: 37.5rem) and (max-width: 62.9375rem) {
  .emotion-1 {
    margin-left: 0%;
  }
}

@media (min-width: 63rem) and (max-width: 79.9375rem) {
  .emotion-1 {
    margin-left: 16.666666666666668%;
  }
}

@media (min-width: 80rem) {
  .emotion-1 {
    margin-left: 33.333333333333336%;
  }
}

@supports (display: grid) {
  .emotion-1 {
    display: block;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 2;
    }
  }

  @media (min-width: 80rem) {
    .emotion-1 {
      grid-template-columns: repeat(12, 1fr);
      grid-column-end: span 12;
      grid-column-start: 5;
    }
  }
}

@media (max-width: 24.9375rem) {
  .emotion-1 {
    margin-top: 0.5rem;
  }
}

@media (min-width: 25rem) {
  .emotion-1 {
    margin-top: 1rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-1 {
    padding: 0 1rem;
  }
}

@media (min-width: 63rem) {
  .emotion-1 {
    padding: 0.5rem 0 0;
    margin-top: 2rem;
  }
}

.emotion-3 {
  margin: 0;
  padding-bottom: 1.5rem;
  width: 100%;
}

.emotion-5 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.emotion-7 {
  border: 0;
  left: 0;
  overflow: hidden;
  position: absolute;
  top: 0;
  width: 100%;
  height: 100%;
}

<div>
  <div
    class="emotion-0 emotion-1 emotion-2"
    dir="ltr"
  >
    <div
      class="emotion-3 emotion-4"
    >
      <div
        class="emotion-5 emotion-6"
        data-e2e="media-player"
      >
        <iframe
          allow="autoplay"
          allowfullscreen=""
          class="emotion-7 emotion-8"
          height="362.953125"
          loading="lazy"
          scrolling="no"
          src="/ws/av-embeds/cps/russian/multimedia/2016/05/160505_v_diving_record/p01kx42v/en-GB"
          title="Media player"
          width="645.25"
        />
        <noscript />
      </div>
    </div>
  </div>
</div>
`;

exports[`MediaPlayer should render the canonical player without a placeholder 1`] = `
.emotion-1 {
  margin-top: 0.5rem;
}

@media (max-width: 14.9375rem) {
  .emotion-1 {
    margin-left: 0%;
  }
}

@media (min-width: 15rem) and (max-width: 24.9375rem) {
  .emotion-1 {
    margin-left: 0%;
  }
}

@media (min-width: 25rem) and (max-width: 37.4375rem) {
  .emotion-1 {
    margin-left: 0%;
  }
}

@media (min-width: 37.5rem) and (max-width: 62.9375rem) {
  .emotion-1 {
    margin-left: 0%;
  }
}

@media (min-width: 63rem) and (max-width: 79.9375rem) {
  .emotion-1 {
    margin-left: 16.666666666666668%;
  }
}

@media (min-width: 80rem) {
  .emotion-1 {
    margin-left: 33.333333333333336%;
  }
}

@supports (display: grid) {
  .emotion-1 {
    display: block;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-1 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 2;
    }
  }

  @media (min-width: 80rem) {
    .emotion-1 {
      grid-template-columns: repeat(12, 1fr);
      grid-column-end: span 12;
      grid-column-start: 5;
    }
  }
}

@media (max-width: 24.9375rem) {
  .emotion-1 {
    margin-top: 0.5rem;
  }
}

@media (min-width: 25rem) {
  .emotion-1 {
    margin-top: 1rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-1 {
    padding: 0 1rem;
  }
}

@media (min-width: 63rem) {
  .emotion-1 {
    padding: 0.5rem 0 0;
    margin-top: 2rem;
  }
}

.emotion-3 {
  margin: 0;
  padding-bottom: 1.5rem;
  width: 100%;
}

.emotion-5 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.emotion-7 {
  border: 0;
  left: 0;
  overflow: hidden;
  position: absolute;
  top: 0;
  width: 100%;
  height: 100%;
}

<div>
  <div
    class="emotion-0 emotion-1 emotion-2"
    dir="ltr"
  >
    <div
      class="emotion-3 emotion-4"
    >
      <div
        class="emotion-5 emotion-6"
        data-e2e="media-player"
      >
        <iframe
          allow="autoplay"
          allowfullscreen=""
          class="emotion-7 emotion-8"
          height="362.953125"
          loading="lazy"
          scrolling="no"
          src="/ws/av-embeds/cps/pidgin/123456789/p01kx42v/en-GB"
          title="Media player"
          width="645.25"
        />
        <noscript />
      </div>
    </div>
  </div>
</div>
`;
